cv-data-service-library/src/main/java/com/trihydro/library/helpers/Utility.java

```diff
@@ -23,325 +19,324 @@
+     * distance to calculate the delta degrees difference between the first two path points
+     * 2) a flat plane assumption was used to calculate the bearing line distance from
+     * the previously calculated latitude and longitude delta distance (d0Latitude, d0Longitude),
+     * and these delta distances provide the flat plane bearing direction to move upstream to
+     * reach the anchor point
+     *
+     * @param firstPoint  The first milepost.
+     * @param secondPoint The second milepost.
+     * @return The anchor coordinate.
+     * @throws IdenticalPointsException If the first and second points are identical, as this would result in a division by zero.
+     */
+    public Coordinate calculateAnchorCoordinate(Milepost firstPoint, Milepost secondPoint)
+        throws IdenticalPointsException {
+        int precision = 9;
+        int metersPerSurfaceDegree = 111195;
+
+        BigDecimal firstPointLatitude =
+            firstPoint.getLatitude().round(new java.math.MathContext(precision));
+        BigDecimal firstPointLongitude =
+            firstPoint.getLongitude().round(new java.math.MathContext(precision));
+        BigDecimal secondPointLatitude =
+            secondPoint.getLatitude().round(new java.math.MathContext(precision));
+        BigDecimal secondPointLongitude =
+            secondPoint.getLongitude().round(new java.math.MathContext(precision));
+
+        // 1) Get the difference in latitude between the first and second points.
         BigDecimal differenceInLatitude = firstPointLatitude.subtract(secondPointLatitude);
 
         // 2) Get the difference in longitude between the first and second points.
-		BigDecimal differenceInLongitude = firstPointLongitude.subtract(secondPointLongitude);
-
-		// 3) Multiply the difference in latitude by the meters per surface degree to get d0Latitude.
-        BigDecimal d0Latitude = new BigDecimal(metersPerSurfaceDegree).multiply(differenceInLatitude);
-        // 4) Multiply the difference in longitude by the cosine of the first point's latitude multiplied by the difference in longitude to get d0Longitude.
-		BigDecimal firstPointLatitudeInRadians = firstPointLatitude.multiply(new BigDecimal(Math.PI)).divide(new BigDecimal(180), new java.math.MathContext(precision));
-		BigDecimal d0Longitude = new BigDecimal(111195).multiply(new BigDecimal(Math.cos(firstPointLatitudeInRadians.doubleValue()))).multiply(differenceInLongitude);
+        BigDecimal differenceInLongitude = firstPointLongitude.subtract(secondPointLongitude);
 
-		// 5) Take the square root of d0Latitude squared plus d0Longitude squared to get d0.
-        BigDecimal d0 = d0Latitude.pow(2).add(d0Longitude.pow(2)).sqrt(new java.math.MathContext(6));
+        // Check if the first and second points are identical
+        if (differenceInLatitude.equals(BigDecimal.ZERO) &&
+            differenceInLongitude.equals(BigDecimal.ZERO)) {
+            throw new IdenticalPointsException();
+        }

```

Comment:
Clarification (non-blocking): This code prevents a division by zero exception at runtime and ensures that callers handle the possibility of identical points.